zeppelin-server/src/main/java/org/apache/zeppelin/service/NotebookService.java

```diff
@@ -1012,7 +1012,28 @@ public void updatePersonalizedMode(String noteId,
   public void moveNoteToTrash(String noteId,
                               ServiceContext context,
                               ServiceCallback<Note> callback) throws IOException {
-    Note note = notebook.getNote(noteId);
+    Note note = null;
+    try {
+       note = notebook.getNote(noteId);
+    } catch (Exception e) {
+        if ((e.getMessage() != null) && (e.getMessage().contains("Fail to parse note json"))) {
```

Comment:
It is better to use specific kind of Exception to represent the corrupted note case.